tests/throttle.cpp

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2018, EPFL/Blue Brain Project
+ *
+ * Responsible Author: Daniel.Nachbaur@epfl.ch
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#define BOOST_TEST_MODULE throttle
+
+#include <boost/test/unit_test.hpp>
+
+#include <brayns/common/Timer.h>
+
+#include "../plugins/RocketsPlugin/Throttle.h"
+
+BOOST_AUTO_TEST_CASE(timeout_not_cleared)
+{
+    brayns::Timeout timeout;
+    bool called = false;
+    timeout.set([&called] { called = true; }, 1);
+    std::this_thread::sleep_for(std::chrono::milliseconds(2));
+    BOOST_CHECK(called);
+}
+
+BOOST_AUTO_TEST_CASE(timeout_with_clear)
+{
+    brayns::Timer timer;
+    timer.start();
+
+    brayns::Timeout timeout;
+
+    bool called = false;
+    timeout.set([&called] { called = true; }, 10000);
+    timeout.clear();
+    BOOST_CHECK(!called);
```

Comment:
Shouldn't the check occur after the time passed?

1. Schedule a callback in 10 seconds
2. clear it
3. Wait for more than 10 sec
4. Check that the callback was not called

Or is the boost check covering that?

Reply:
Will add this afterwards. Not 10 seconds then of course